src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -29,6 +29,42 @@
+        Reads ``ExecutionConfig.max_steps`` when set (single knob honoured by the
+        OpenAI-native and LiteLLM tool-execution loops); otherwise falls back to
+        ``max_iter`` and finally ``default`` for backward compatibility.
+        """
+        execution = getattr(self, "execution", None)
+        if execution is not None:
+            resolver = getattr(execution, "resolved_max_steps", None)
+            if callable(resolver):
+                try:
+                    return int(resolver())
+                except Exception:
+                    pass
```

Ruff (0.15.20)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)